### ISSUE TITLE
Roslyn suggests conflicting IDE0044/IDE0064 code fixes

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn#59577]
**Version Used**: VS2022 17.1

**Steps to Reproduce**:

1. Paste the following code:

```csharp
public struct Foo
{
    private readonly IntPtr x;

    public Foo(IntPtr x) => this.x = x;

    public void Bar()
    {
        this = default;
    }
}
```

2. Roslyn will show IDE0044 "make field readonly" on `x`
3. Do that
3. Roslyn will show IDE0064 "struct contains assignments to this outside of constructor, make fields writeable"
4. Go to step 2.

**Expected Behavior**:

I'd expect Roslyn to either not show IDE0044, or at least decide between the two 😆

**Actual Behavior**:

Unless you ignore the suggestions, you might be stuck applying them for quite a while here 😄

**Comment by sharwell** [comment]
IDE0044 (Make field readonly) should not be reported here.
